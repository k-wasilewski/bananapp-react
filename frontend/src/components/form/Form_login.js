import React from 'react'
import { Formiz, useForm } from '@formiz/core'
import { isEmail } from '@formiz/validations' // Import some validations
import { Form_field } from './Form_field'
import axios from "axios";
import {Redirect} from "react-router-dom";
import { useState } from 'react'

//LOGIN

export const Form_login = () => {

    const myForm = useForm();

    const [redirect, setRedirect] = useState(0);

    const handleSubmit = (values, event) => {
        axios.post('http://localhost:8081',
            "username=" + values.email + "&" + "password=" + values.password
        ).then(function (response) {
            if (response.status === 200) {
                setRedirect(response.data);
            }
        });
        axios.post('http://localhost:8082/auth/user',
            "uname=" + values.email
        ).then(function (response) {
            if (response.status === 200) {
                setRedirect(response.data);
            }
        });
    };

    if (redirect == 0) {
        return (
            <Formiz
                connect={myForm}
                onValidSubmit={handleSubmit}
            >
                <form
                    noValidate
                    onSubmit={myForm.submit}
                >
                    <Form_field
                        name="email"
                        label="E-mail: "
                        validations={[
                            {
                                rule: isEmail(),
                                message: 'This is not a valid email',
                            },
                        ]}
                    />
                    <Form_field
                        name="password"
                        label="Password: "
                        type="password"
                    />
                    <button
                        type="submit"
                        disabled={!myForm.isValid}
                    >
                        Submit
                    </button>
                </form>
            </Formiz>
        );
    } else if (redirect == "success") {
        return (<Redirect to='/success' />);
    } else {
        return (
            <Formiz
                connect={myForm}
                onValidSubmit={handleSubmit}
            >
                <form
                    noValidate
                    onSubmit={myForm.submit}
                >
                    Login failed
                    <Form_field
                        name="email"
                        label="E-mail: "
                        validations={[
                            {
                                rule: isEmail(),
                                message: 'This is not a valid email',
                            },
                        ]}
                    />
                    <Form_field
                        name="password"
                        label="Password: "
                        type="password"
                    />
                    <button
                        type="submit"
                        disabled={!myForm.isValid}
                    >
                        Submit
                    </button>
                </form>
            </Formiz>
        );
    }
};
