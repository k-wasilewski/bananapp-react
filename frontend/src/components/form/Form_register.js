import React from 'react'
import { Formiz, useForm } from '@formiz/core'
import { isEmail } from '@formiz/validations' // Import some validations
import { Form_field } from './Form_field' // Import your field
import axios from "axios";
import { useState } from 'react'

//REGISTER

export const Form_register = () => {
    const myForm = useForm();
    const [redirect, setRedirect] = useState(0);

    const handleSubmit = (values) => {
        axios.post('http://localhost:8081/create-user',
            "username=" + values.email + "&" + "password=" + values.password
        ).then(function (response) {
            if (response.status === 200) {
                setRedirect(response.data);
            }
        });
    };

    let message = (<div/>);
    if (redirect == 'success') {
        message = (<div>Registration success</div>)
    } else if (redirect != 0) {
        message = (<div>E-mail already exists</div>)
    }

    return (
        <Formiz
            connect={myForm}
            onValidSubmit={handleSubmit}
        >
            {message}
            <form
                noValidate
                onSubmit={myForm.submit}
            >
                <Form_field
                    name="email"
                    label="E-mail: "
                    validations={[
                        {
                            rule: isEmail(),
                            message: 'This is not a valid email',
                        },
                    ]}
                />
                <Form_field
                    name="password"
                    label="Password: "
                    type="password"
                />
                <Form_field
                    name="passwordConfirm"
                    label="Confirm password: "
                    type="password"
                    validations={[
                        {
                            rule: (value) => myForm.values.password === value,
                            deps: [myForm.values.password],
                            message: 'Passwords do not match',
                        }
                    ]}
                />
                <button
                    type="submit"
                    disabled={!myForm.isValid}
                >
                    Submit
                </button>
            </form>
        </Formiz>
    );
};

