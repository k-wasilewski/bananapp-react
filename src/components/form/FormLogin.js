import React from 'react'
import { Formiz, useForm } from '@formiz/core'
import { isEmail } from '@formiz/validations' // Import some validations
import { FormField } from './FormField'
import axios from 'axios';
import {Redirect} from 'react-router-dom';
import { useState } from 'react'

export const FormLogin = () => {

    const myForm = useForm();

    const [redirect, setRedirect] = useState(0);
    const [username, setUsername] = useState(0);

    const handleSubmit = (values) => {
        axios.post('http://localhost:8081',
            `username=${values.email}&password=${values.password}`,
            {withCredentials: true}
        ).then(function (response) {
            if (response.data === 'success') {
                setUsername(values.email);
                setRedirect('success');
            } else {
                setRedirect('fail');
            }
        });
        axios.post('http://localhost:8082/auth/user',
            'uname=' + values.email
        );
    };

    const errorMsg = (redirect !== 'success' && redirect !== 0) ?
        (<div>Login failed</div>)
        :
        (<div/>);

    if (redirect !== 'success') {
        return (
            <Formiz connect={myForm} onValidSubmit={handleSubmit}>
                <form noValidate onSubmit={myForm.submit}>
                    {errorMsg}
                    <FormField name='email' label='E-mail: ' validations=
                        {[{
                                rule: isEmail(),
                                message: 'This is not a valid email',
                            }]}
                    />
                    <FormField name='password' label='Password: ' type='password' />
                    <button className='formBtn' type='submit' disabled={!myForm.isValid}>
                        Submit
                    </button>
                </form>
            </Formiz>
        );
    } else if (redirect === 'success') {
        return (<Redirect to={{
            pathname: '/success',
            state: {username: username}}}/>
        );
    }
};

