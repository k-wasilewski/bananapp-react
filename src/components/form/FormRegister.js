import React from 'react'
import { Formiz, useForm } from '@formiz/core'
import { isEmail } from '@formiz/validations' // Import some validations
import { FormField } from './FormField' // Import your field
import axios from 'axios';
import { useState } from 'react'

export const FormRegister = () => {
    const myForm = useForm();
    const [redirect, setRedirect] = useState(0);

    const handleSubmit = (values) => {
        axios.post('http://localhost:8081/create-user',
            `username=${values.email}&password=${values.password}`
        ).then(function (response) {
            if (response.status === 200) {
                setRedirect(response.data);
            }
        });
    };

    let message = (<div/>);
    if (redirect === 'success') {
        message = (<div>Registration success</div>);
    } else if (redirect !== 0) {
        message = (<div>Registration failed</div>);
    }

    return (
        <Formiz connect={myForm} onValidSubmit={handleSubmit} >
            {message}
            <form noValidate onSubmit={myForm.submit}>
                <FormField name='email' label='E-mail: ' validations=
                    {[{
                            rule: isEmail(),
                            message: 'This is not a valid email',
                        }]}
                />
                <FormField name='password' label='Password: ' type='password' />
                <FormField name='passwordConfirm' label='Confirm password: ' type='password'
                    validations=
                        {[{
                            rule: (value) => myForm.values.password === value,
                            deps: [myForm.values.password],
                            message: 'Passwords do not match',
                        }]}
                />
                <button className='formBtn' type='submit' disabled={!myForm.isValid}>
                    Submit
                </button>
            </form>
        </Formiz>
    );
};

