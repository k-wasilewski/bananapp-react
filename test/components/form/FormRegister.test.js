import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../../src/redux/store';
import {BrowserRouter} from "react-router-dom";
import {FormRegister} from "../../../src/components/form/FormRegister";
import {Provider} from "react-redux";
import {configure} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {mount} from "enzyme";
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { act } from "react-dom/test-utils";
import { fireEvent } from "@testing-library/react";

describe("FormRegister rendering specification", () => {
    it('FormRegister is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <FormRegister/>
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("FormRegister functional specification", () => {
    let component;

    beforeEach(() => {
        configure({adapter: new Adapter()});
    });

    afterEach(() => {
        component.unmount();
    });

    it('renders success msg when state value redirect equals "success"', (done) => {
        const mock = new MockAdapter(axios);
        const okResp = 'success';
        mock.onPost().reply(200, okResp);

        component = mount(
            <Provider store={store}>
                <FormRegister/>
            </Provider>
        );

        const email = "test@test.pl";
        const password = "test";
        component.find('#emailField').simulate('change', { target: { value: email } });
        component.find('#passwordField').simulate('change', { target: { value: password } });
        act(() => {
            component.find('#form').at(0).simulate('submit');
        });
        component.update();

        setTimeout(function () {
            expect(component.find('#msg').html())
                .toContain('Registration success');
            done();
        }, 500);
    });

    it('renders fail msg when state value redirect doesnt equal "success"', (done) => {
        const mock = new MockAdapter(axios);
        const failResp = 'fail';
        mock.onPost().reply(200, failResp);

        component = mount(
            <Provider store={store}>
                <FormRegister/>
            </Provider>
        );

        const email = "test@test.pl";
        const password = "test";
        component.find('#emailField').simulate('change', { target: { value: email } });
        component.find('#passwordField').simulate('change', { target: { value: password } });
        act(() => {
            component.find('#form').at(0).simulate('submit');
        });
        component.update();

        setTimeout(function () {
            expect(component.find('#msg').html())
                .toContain('Registration failed');
            done();
        }, 500);
    });
});