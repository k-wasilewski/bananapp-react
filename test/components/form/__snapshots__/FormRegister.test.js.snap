// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRegister rendering specification FormRegister is rendered 1`] = `
Array [
  <div
    id="msg"
  >
    
  </div>,
  <form
    id="form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="demo-form-group "
    >
      <label
        className="demo-label"
        htmlFor="formiz-field-id-2zycffqf7"
      >
        E-mail: 
      </label>
      <input
        aria-describedby={null}
        aria-invalid={false}
        aria-required={false}
        className="demo-input"
        id="formiz-field-id-2zycffqf7"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="demo-form-group "
    >
      <label
        className="demo-label"
        htmlFor="formiz-field-id-64kfsf7zu"
      >
        Password: 
      </label>
      <input
        aria-describedby={null}
        aria-invalid={false}
        aria-required={false}
        className="demo-input"
        id="formiz-field-id-64kfsf7zu"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="demo-form-group "
    >
      <label
        className="demo-label"
        htmlFor="formiz-field-id-ll2iz0zgj"
      >
        Confirm password: 
      </label>
      <input
        aria-describedby={null}
        aria-invalid={false}
        aria-required={false}
        className="demo-input"
        id="formiz-field-id-ll2iz0zgj"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="formBtn"
      disabled={false}
      id="submit"
      type="submit"
    >
      Submit
    </button>
  </form>,
]
`;
